### PR TITLE
Improvement for pspm_extract_segments

### PR DESCRIPTION
**Pull request by irojkov-ph**
Here I want to propose a Improvement for `pspm_extract_segments` function which will be very useful for the feature request #50 and could be also for other future functions.

Changes proposed in this pull request:
- The main change is that now with these changes one can use the functions as following:
``` [sts, segments] = pspm_extract_segments('manual', data_raw, sr, timing, options)```
where `data_raw` would be a numeric array of a cell array of numeric arrays of raw data and `sr`  the sampling rate of the data (or an array of sampling rates of data).
- The test for this function will not work because of the line n° 184 which is now a valid input for the function. Thus it has to be replaced by the following lines:
```
this.verifyWarning(@() pspm_extract_segments('manual',struct('a',10),0,timing), 'ID:invalid_input');
this.verifyWarning(@() pspm_extract_segments('manual',[1,3],logical(32),timing), 'ID:invalid_input');
```
All the rest of the tests stays valid.
- I have tested the new function with my data and it reproduces the result in an identical way as when it was used with the previous version of `pspm_extract_segments`.

### CODE REVIEW

**Comment by irojkov-ph** [comment]
In the last commit I've change the function such that "timeunit" should be seconds when raw data are given to it. Obviously markers and other time unit wouldn't make sense since we do not have any filename.

**Comment by irojkov-ph** [comment]
An update of this pull request:
- I've made some changes in the original code (typically on lines 569-587) because when making deeper test of the actual function in the manual mode w/o raw data, I've realized that there are some errors. Thus I've corrected it.
- Secondly, upon reflection and after discussing it with @fmelinscak, I've changed my initial implementation. Now one can choose any of the three types of time units when using this function with raw data.

File: src/pspm_extract_segments.m [hunk 1]
```diff
@@ -1,680 +1,738 @@
...
+    %                           will be treated as session. Onset values are
+    %                           averaged through conditions and sessions.
+    %       data_raw:           Numeric raw data or a cell array of numeric raw data.
+    %       chan:               Channel number or cell of channel numbers which
+    %                           defines which channel should be taken to
+    %                           extract the segments. Chan should correspond to
+    %                           data_fn and should have the same length. If
+    %                           data_fn is a cell and chan is a single number,
+    %                           the number will be taken for all files.
+    %       sr:                 Array of sampling rates (same dimension as data_raw)
```
Comment (eozd) [inline src/pspm_extract_segments.m]: Writing "same dimension as `data_raw`" might be confusing, so we should note that this is the case only if `data_raw` is a cell array.

File: src/pspm_extract_segments.m [hunk 2]
```diff
@@ -1,680 +1,738 @@
...
+    if ~isfield(options, 'outputfile')
+        options.outputfile = '';
+    end;
+    
+    % overwrite
+    if ~isfield(options, 'overwrite')
+        options.overwrite = 0;
+    end;
+    
+    % dont_ask_overwrite
```
Comment (eozd) [inline src/pspm_extract_segments.m]: (Low priority) Now I see that we have this undocumented previously implemented "don't ask overwrite" option for some reason, and we don't need it. Could you remove everything related to this option and change line 715 to `else`?

**Comment by eozd** [comment]
Also, could you fix the failing test? If this is related to the comment you made about tests, please do change the tests in this pull request as well so that everything passes; afterwards I will approve it.

File: src/pspm_extract_segments.m (hunk 1 above)
Comment (irojkov-ph) [reply src/pspm_extract_segments.m]: Copy that

File: src/pspm_extract_segments.m (hunk 2 above)
Comment (irojkov-ph) [reply src/pspm_extract_segments.m]: Copy that

**Pull request merged**